source/parser.hera

```diff
@@ -518,7 +518,12 @@ TrailingDeclaration
 # https://262.ecma-international.org/#prod-ConciseBody
 FatArrowBody
   # If same-line single expression, avoid wrapping in braces
-  !EOS NonPipelinePostfixedExpression:exp !TrailingDeclaration !SemicolonDelimiter -> exp
+  !EOS NonPipelinePostfixedExpression:exp !TrailingDeclaration !SemicolonDelimiter ->
+    // Ensure object literal is wrapped in parens
+    if (exp.type === "ObjectExpression") {
```

Comment:
I think it would be better to use the new `makeExpressionStatement` helper, which also covers more cases.

Reply:
I don't think that's necessary since anonymous functions are fine here. I don't think comma expressions can get picked up here either, they'd be part of the outer expression.

Reply:
Ah, good point. I thought anonymous functions were an issue, but they're not.